keps/sig-release/1731-publishing-packages/README.md

```diff
@@ -149,51 +183,236 @@ Scenario: [...]
+  (e.g. **`isv:kubernetes:core:stable:v1.26`**)
+  - We'll publish our packages from this project
+- Each **`${k8s_release}`** has a **building** subproject as a subproject of the appropriate publishing subproject
+  (e.g. **`isv:kubernetes:core:stable:v1.26:stage`**)
+  - We'll build packages in this project
+  
+Having **`${k8s_release}`** subproject as a subproject of **`${channel}`** is required so we can build multiple
+releases in parallel. This is because if changes are pushed to the package, the ongoing build process is aborted.
+Running builds sequentially is not an option because that would slow down the release process too much.
+
+Mentioned subprojects can be created manually via the web interface. Upon creating appropriate **`${k8s_release}`**
+subprojects, the target operating systems and architectures (listed earlier) must be configured for those subprojects
+(via the Repositories option). Additionally, some meta configuration is needed to declare the projects as publishing
+and building. This is to be done by the Release Managers before cutting the first alpha release for that minor release.
+The concrete configuration steps will be documented outside of this KEP, as part of the Release Managers Handbook.
+We'll also consider automating this in some form.
+
+#### Packages in OBS
+
+**Package** object must be created in the **`${k8s_release}`** **building** subproject for each package that we want
+to build and publish. This can be done via the `osc` command-line tool or the web interface. The created package 
+inherits information about the target operating systems and architectures from the subproject.
+
+Creating packages is to be done by the Release Managers before cutting the first alpha release for that minor release.
+We'll consider automating this in some form.
+
+Packages in the **publishing** subproject are created automatically for each published build. Those automatically
+packages are named as `<package-name>.<timestamp>`, e.g. `kubectl.20230120135613`. This naming schema doesn't affect
+package managers or users, i.e. those packages are still installable by their original name
+(e.g. `apt install kubectl=1.26.0*`). More about building and publishing packages is explained in the next sections.
+
+#### Package Sources
+
+We'll build packages using pre-built binaries instead of pushing sources and then building binaries in the OBS pipeline.
+The reasoning for this is:
+
+- We already have our own release pipeline. Adding another release pipeline would increase the maintenance burden for
+  Release Managers
+- It would increase the effort for updating build dependencies such as Go
+- It would increase the effort for validating correctness of created binaries
+- Binary published by our release process would differ to binaries built by OBS
+  - Additional efforts would be needed to get reproducible builds working
+  - We would also lose cosign signatures for binaries built by OBS
```

Comment:
> lose cosign signatures for binaries built by OBS

I’m assuming the answer is no, but wouldn’t repackaging binaries, outside the release process, “void” the integrity of cosign anyway?

Alternately, what level of assurance does SUSE provide for the builds/build environments? How will we validate the integrity of our releases via OBS?

(This might be what @aojea is referring to?)

Reply:
> I’m assuming the answer is no, but wouldn’t repackaging binaries, outside the release process, “void” the integrity of cosign anyway?

OBS (our spec files to be precise) is just going to reuse the prebuilt binaries. Given that, I think that installed binaries should preserve the signature and checksums, but I need to double-check that.

> Alternately, what level of assurance does SUSE provide for the builds/build environments?

@darix @dirkmueller Can you please provide some feedback on this?

> How will we validate the integrity of our releases via OBS?

I think we can create some job that, let's say, installs the package and then we verify that installed binary is correct (checksums are correct and signature is in the place). I'll double-check that.

Reply:
The only potential tampering would be the debuginfo package generation logic, which we can trivially disable in the spec file (and I think we already have).

Reply:
You can certainly verify  the checksum of the files if all you're doing is repackaging binaries.

For the build process, OBS conforms to SLSA level 4 (https://github.com/slsa-framework/slsa/), i.e. all builds are hermetic, isolated, parameterless...

Having said this, the build environment (i.e. the path you have in the project config) is usually some distribution, so you're only as save as the packages from the distribution. I.e. you need to trust:

- the hardware OBS uses
- the software that runs the build (sets up the VM and the like). This is https://github.com/openSUSE/obs-build/
- the distribution you use for the build environment